src/azure-cli/azure/cli/command_modules/backup/custom_base.py

```diff
@@ -160,13 +180,57 @@ def list_associated_items_for_policy(client, resource_group_name, vault_name, na
     return custom.list_associated_items_for_policy(client, resource_group_name, vault_name, name)
 
 
+def list_protectable_items(cmd, client, resource_group_name, vault_name, workload_type, container_name=None):
+    container_uri = None
+    if container_name:
+        if custom_help.is_native_name(container_name):
+            container_uri = container_name
+        else:
+            container_client = backup_protection_containers_cf(cmd.cli_ctx)
+            container = show_container(cmd, container_client, container_name, resource_group_name, vault_name,
+                                       "AzureWorkload")
+            custom_help.validate_container(container)
```

Comment:
Why not validate using [validator](https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#validators) for argument?

Reply:
Argument is a string i.e. container_name. We validate it by doing a GET call and checking if the container actually exists. Hence it is not a validator for **argument**. Any valid string can be passed as an argument. It is better to make it a part of logic rather than adding it as a validator for argument.